src/main/java/touro/GameOfLife/Grid.java

```diff
@@ -1,4 +1,77 @@
 package touro.GameOfLife;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
 public class Grid {
+    private Direction[] directions = Direction.values();
+
+    //TODO: find a reasonable height and width
+    public static final int HEIGHT = 20;
+    public static final int WIDTH = 20;
+
+    private final List<Cell> allCells = new ArrayList<>();
+    private final HashSet<Cell> liveCells = new HashSet<>();
+
+    public Grid() {
+        for (int x = 0; x < WIDTH; x++) {
+            for (int y = 0; y < HEIGHT; y++) {
+                allCells.add(new Cell(x,y));
+            }
+        }
+    }
+
+    public List<Cell> getAllCells() {
+        return allCells;
+    }
+
+    //TODO: will the view need this as a list?
+    public HashSet<Cell> getLiveCells() {
+        return liveCells;
+    }
+
+    /***
+     * @param cell
+     * @return false if cell is already in liveCells
+     */
+    public boolean addCellToLiveCells(Cell cell) {
+        return liveCells.add(cell);
+    }
+
+    /***
+     * @param cell
+     * @return true if cell was in liveCells
+     */
+    public boolean removeCellFromLiveCells(Cell cell) {
+        return liveCells.remove(cell);
+    }
+
+    public void advance() {
+        HashSet<Cell> newLiveCells = new HashSet<>();
+
+        for (Cell cell : allCells) {
+            int liveNeighbors = 0;
+
+            for (Direction direction : directions) {
+                Cell neighbor = new Cell(cell.moveTo(direction));
+
+                if (liveCells.contains(neighbor)) {
```

Comment:
this isn't working

Reply:
Maybe because you never added cells to liveCells?

Reply:
I did make some tests to try it out though, and it still did not work even with liveCells has things in it

Reply:
I searched and found: "If you look at the documentation for List.Contains it states that it uses the default Equals method for the object. If you haven't written an Equals method for your class then it will default to checking if they are equal by seeing if they are the exact same instance of the object."
So I'm guessing that for some reason your equals method is not working or not being recognized. 
Perhaps it's because the parameter needs to be Object?
That is just a guess.

Reply:
There's an old saying, "If all your have is a hammer, everything looks like a nail."

`moveTo()` was a very specific method for a very specific purpose. Is that really necessary here? Or are you only doing it because you saw it in Snake?

Reply:
I believe that we do still need it. Perhaps syntactically we are not "moving", but this creates a cell in the x,y position moved in that direction. This is the way that the code can compare things to a cell's neighbor, unless I am misunderstanding.

Reply:
It turns out the problem I was having was that the `hashCode()` method needed to be overridden (in addition to the `equals()` method). Once that was added in, everything works as expected.